File: Behavioral/ChainOfResponsibility/JavaScript/basicSupport.js
// BasicSupport.js

const SupportHandler = require("./supportHandler");

class BasicSupport extends SupportHandler {
  handleRequest(request) {
    if (request === "Basic Issue") {
      console.log(`BasicSupport: Handling request - ${request}`);
    } else if (this.nextHandler) {
      this.nextHandler.handleRequest(request);
    } else {
      console.log(`No handler available for - ${request}`);
    }
  }
}

module.exports = BasicSupport;

File: Behavioral/ChainOfResponsibility/Java/BasicSupport.java
// BasicSupport.java

public class BasicSupport extends SupportHandler {
  @Override
  public void handleRequest(String request) {
    if (request.equals("Basic Issue")) {
      System.out.println("BasicSupport: Handling request - " + request);
    } else if (nextHandler != null) {
      nextHandler.handleRequest(request); // Only pass if not handled
    } else {
      System.out.println("No handler available for - " + request);
    }
  }
}
